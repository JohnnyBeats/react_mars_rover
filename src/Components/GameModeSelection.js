import React from "react";
import GameModeEnum from "../Classes/GameModeEnum";

class GameModeSelection extends React.Component {
    constructor(props) {
        super(props);
        this.setGameMode = this.setGameMode.bind(this);
    }

    setGameMode(event) {
        this.props.setGameMode(event.target.value);
    }

    render() {
        return(
            <div className="gameModeSelection">
                <button data-testid="Multiplayer" className="gameMode button" value={GameModeEnum.MULTIPLAYER} onClick={this.setGameMode}>
                    Multiplayer
                </button>
                <br/>
                <br/>
                <button data-testid="SinglePlayer" className="gameMode button" value={GameModeEnum.SINGLE_PLAYER} onClick={this.setGameMode}>
                    Single Player
                </button>
            </div>
        )
    }
}

export default GameModeSelection;