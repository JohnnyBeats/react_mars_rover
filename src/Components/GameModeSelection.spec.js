import React from 'react';
import {fireEvent, render} from "@testing-library/react";
import GameModeSelection from "./GameModeSelection";

describe('The GameModeSelection component', ()=> {
    const setup = () => {
        const setGameModeMock = jest.fn();
        const utils = render(<GameModeSelection setGameMode={setGameModeMock}/>);
        const MultiplayerButton = utils.getByTestId('Multiplayer');
        const SinglePlayerButton = utils.getByTestId('SinglePlayer');

        return {
            MultiplayerButton,
            SinglePlayerButton,
            setGameModeMock,
            ...utils,
        }
    };

    it('renders two mode buttons without regression', ()=>{
        const {container} = setup();

        expect(container.firstChild).toMatchSnapshot();
    });

    it('calls setGameMode once on multiplayer click with value Multiplayer', ()=>{
        const {MultiplayerButton, setGameModeMock} = setup();

        fireEvent.click(MultiplayerButton);

        expect(setGameModeMock).toHaveBeenCalledTimes(1);

        expect(setGameModeMock).toHaveBeenCalledWith("Multiplayer");
    });

    it('calls setGameMode once on single player click with value SinglePlayer', ()=>{
        const {SinglePlayerButton, setGameModeMock} = setup();

        fireEvent.click(SinglePlayerButton);

        expect(setGameModeMock).toHaveBeenCalledTimes(1);
        expect(setGameModeMock).toHaveBeenCalledWith("SinglePlayer");
    });
});