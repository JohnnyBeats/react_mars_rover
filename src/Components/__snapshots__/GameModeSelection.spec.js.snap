// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GameModeSelection component renders two mode buttons without regression 1`] = `
<div
  class="gameModeSelection"
>
  <button
    class="gameMode button"
    data-testid="Multiplayer"
    value="Multiplayer"
  >
    Multiplayer
  </button>
  <br />
  <br />
  <button
    class="gameMode button"
    data-testid="SinglePlayer"
    value="SinglePlayer"
  >
    Single Player
  </button>
</div>
`;
